### PR TITLE
fix: retry and keep alive

### PR DESCRIPTION
**Pull request by ddubuc**
Need to re-add features to manage retry and keep alive in the tcp connection

### CODE REVIEW

File: packages/csharp/ArmoniK.Api.Client/Options/GrpcClient.cs [hunk 1]
```diff
@@ -72,10 +72,41 @@ public class GrpcClient
...
+
+    /// <summary>
+    ///   KeepAliveTimeInterval is the interval in milliseconds at which the connection will be kept alive.
+    /// </summary>
+    public int KeepAliveTimeInterval { get; set; } = 30000;
+
+    /// <summary>
+    ///   MaxIdleTime is the maximum idle time in minutes after which the connection will be closed.
+    /// </summary>
+    public int MaxIdleTime { get; set; } = 5;
```
Comment (lemaitre-aneo) [inline packages/csharp/ArmoniK.Api.Client/Options/GrpcClient.cs:95]: You could use TimeSpan for those instead of int in milliseconds.

**Review by aneojgurhem — APPROVED** [review]
LGTM

**Review by dbrasseur-aneo — APPROVED** [review]
Lgtm, but is the Grpc.Core channel the only one needing that ?

**Comment by lemaitre-aneo** [comment]
> Lgtm, but is the Grpc.Core channel the only one needing that ?

In fact, this fix is *not* for Grpc.Core. I will try to do that in another PR, but the options are clumsy.

**Pull request merged**